pypeit/metadata.py

```diff
@@ -382,7 +382,10 @@ def find_frames(self, ftype, sci_ID=None, index=False):
             return self['framebit'] == 0
         indx = self.bitmask.flagged(self['framebit'], ftype)
         if sci_ID is not None:
-            indx &= (self['sci_ID'] & sci_ID > 0)
+            if ftype == 'science':  # THIS IS KEY FOR near-IR where arc=science
+                indx &= (self['sci_ID'] == sci_ID)
+            else:
+                indx &= (sci_ID > 0)
         return np.where(indx)[0] if index else indx
```

Comment:
I'm not following this line. NIRES has been reducing just fine without this line, and there the science is also the arc.

Reply:
Hmm that's interesting. When I was running PypeIt on my NIRSPEC data, it would re-reduce the one science frame that is also the arc N times, where N is the total number of raw science frames.